src/fairness/weighted_tree/test/load_accounting_db.cpp

```diff
@@ -0,0 +1,302 @@
+                sqlite3_column_text (b_select_associations_stmt, 3));
+
+            // add user as a child of the node
+            user_node = std::make_shared<weighted_tree_node_t> (
+                            parent_bank,
+                            username,
+                            true,
+                            std::stoll (user_shares),
+                            std::stoll (user_job_usage));
+            node->add_child (user_node);
+            leaf_bank_usage += std::stoi (user_job_usage);
+
+            rc = sqlite3_step (b_select_associations_stmt);
+        }
+        if (rc == SQLITE_ERROR) {
+            fprintf (stderr, "%s\n", sqlite3_errmsg (DB));
+            sqlite3_close (DB);
+
+            return nullptr;
+        }
+        // aggregate usage from user nodes up to their respective banks
+        // and up to the root bank
+        for (std::shared_ptr<weighted_tree_node_t> curr_ancestor = node;
+             curr_ancestor != nullptr;
+             curr_ancestor = curr_ancestor->get_parent ()) {
+
+            curr_ancestor->set_usage (curr_ancestor->get_usage ()
+                                      + leaf_bank_usage);
+        }
+    }
+    // otherwise, this bank has sub banks, so call this
+    // function again with the first sub bank it found
+    else {
+        parent_bank = node;
+        while (rc == SQLITE_ROW) {
+            // execute SQL statement
+            std::string bank = reinterpret_cast<char const *> (
+                sqlite3_column_text (b_select_sub_banks_stmt, 0));
+            banks.push_back (bank);
+            rc = sqlite3_step (b_select_sub_banks_stmt);
+        }
+        if (rc == SQLITE_ERROR) {
+            fprintf (stderr, "%s\n", sqlite3_errmsg (DB));
+            sqlite3_close (DB);
+
+            return nullptr;
+        }
+        for (std::string b : banks) {
+            // reset the prepared statements back to their initial state and
+            // clear their bindings
+            sqlite3_clear_bindings (b_select_associations_stmt);
+            sqlite3_reset (b_select_associations_stmt);
+            sqlite3_clear_bindings (b_select_sub_banks_stmt);
+            sqlite3_reset (b_select_sub_banks_stmt);
+            sqlite3_clear_bindings (b_select_shares_stmt);
+            sqlite3_reset (b_select_shares_stmt);
+            get_sub_banks (DB,
+                           b,
+                           parent_bank,
+                           b_select_shares_stmt,
+                           b_select_sub_banks_stmt,
+                           b_select_associations_stmt);
+        }
+    }
+    return root_ptr;
+}
+
+std::shared_ptr<weighted_tree_node_t>
+load_accounting_db (const std::string &path) {
+    // SQL statements to retrieve data from flux-accounting database
+    sqlite3_stmt *b_select_root_bank_stmt;
+    sqlite3_stmt *b_select_shares_stmt;
+    sqlite3_stmt *b_select_sub_banks_stmt;
+    sqlite3_stmt *b_select_associations_stmt;
+
+    sqlite3 *DB;
```

Comment:
I think it would be a good practice to add `nullptr` to these since you pass them into functions to get the results. Setting the initial condition would help for debugging.